### PR TITLE
GDS 110 | use Jupyter password, not auth token

### PR DESCRIPTION
**Pull request by recursive-automata**
## Change Notes
* Added a script for generating necessary Jupyter notebook configuration object.
* Added a Make rule to create Jupyter notebook configuration file.
* Added a Make rule to start Jupyter in Docker and open a tab in Chrome 
* Modified Dockerfile to include the Jupyter configuration.

## Jira Issues
- [GDS-110](https://55places.atlassian.net/browse/GDS-110)

## Testing
Works on my machine. ¯\\\_(ツ)\_/¯

## Implementation
* Software Product

### CODE REVIEW

File: {{ cookiecutter.repo_name }}/Makefile [hunk 1]
```diff
@@ -135,6 +138,17 @@ run_docker: populate_aws_id
...
+## Run Jupyter in a container and hit it in Chrome
+start_jupyter: populate_aws_id
+	docker run -dt -p 7777:8888 \
+            -e "AWS_ACCOUNT_ID=$(AWS_ACCOUNT_ID)" \
+            -e "AWS_ACCESS_KEY_ID=$(shell aws --profile default configure get aws_access_key_id)" \
+            -e "AWS_SECRET_ACCESS_KEY=$(shell aws --profile default configure get aws_secret_access_key)" \
+            --name=$(PROJECT_NAME) $(PROJECT_NAME):latest \
+	    jupyter notebook --ip 0.0.0.0 --allow-root \
+	&& sleep 1 \
+        && /usr/bin/open -a "/Applications/Google Chrome.app" 'http::/localhost:7777'
```
Comment (davidzornek) [inline {{ cookiecutter.repo_name }}/Makefile:152]: Nice touch opening Chrome as part of the command. Couldn't possibly be more convenient.  With that, maybe even I will run my notebooks within the container!

**Review by davidzornek — APPROVED** [review]

File: {{ cookiecutter.repo_name }}/Makefile [hunk 2]
```diff
@@ -120,8 +120,11 @@ pull_base_image: docker_login populate_aws_id
 	docker pull ${AWS_BASE_IMAGE} \
         && docker tag ${AWS_BASE_IMAGE} analytics-base:${IMAGE_VERSION}
 
+jupyter_notebook_config.json:
+	$(PYTHON_INTERPRETER) create_notebook_config.py
```
Comment (jillrobinson) [inline {{ cookiecutter.repo_name }}/Makefile]: Could we please add a comment saying that you'll be requested to set a password here (for jupyter notebook)? And perhaps a note that you will need to remember that password for whenever you launch the container and jupyter?

Comment (jillrobinson) [reply {{ cookiecutter.repo_name }}/Makefile]: Could we make a JIRA ticket to update the README to reflect this process?

**Review by jillrobinson — APPROVED** [review]
Just a documentation request:

File: {{ cookiecutter.repo_name }}/Makefile (hunk 2 above)
Comment (davidzornek) [reply {{ cookiecutter.repo_name }}/Makefile]: I had that thought as well, but for some reason didn't say it. Upon revisiting that thought, I incline toward saying that updating the readme is in scope for this PR.

Comment (davidzornek) [reply {{ cookiecutter.repo_name }}/Makefile]: Changing my approval to a change request to reflect my change of mind about the readme.

File: README.md [hunk 3]
```diff
@@ -67,25 +56,25 @@ The directory structure of your new project looks like this:
...
+│   │                     predictions
 │   └── visualization  <- Scripts to create exploratory and results oriented visualizations
-│       └── visualize.py
 │
+├── create_notebook_config.py <- script for creating Jupyter notebook configuration.
 └── tox.ini            <- tox file with settings for running tox; see tox.testrun.org
 ```
 
-## Contributing
+### Useful Make rules
```
Comment (davidzornek) [inline README.md:70]: Approve with one change (you don't need to get another review before merging):
It's probably worth mentioning that simply typing `make` will list all available commands (but it excludes those rules associated with running the analysis; applies only to technical commands like this).

**Review by davidzornek — APPROVED** [review]

**Pull request merged**